DSCResources/MSFT_xIPAddress/MSFT_xIPAddress.psm1

```diff
@@ -252,41 +209,32 @@ function ValidateProperties
                     PrefixLength = $SubnetMask
                     InterfaceAlias = $InterfaceAlias
                 }
-                if($DefaultGateway)
+
+                # Use $AddressFamily to select the IPv4 or IPv6 destination prefix
+                $DestinationPrefix = "0.0.0.0/0"
+                if($AddressFamily -eq "IPv6")
                 {
-                    $Parameters['DefaultGateway'] = $DefaultGateway
+                    $DestinationPrefix = "::/0"
                 }
 
-                if(-not $NoReset)
+                # Get all the default routes - this has to be done in case the IP Address is
+                # beng Removed
+                $defaultRoutes = Get-NetRoute -InterfaceAlias $InterfaceAlias -AddressFamily `
+                    $AddressFamily -ErrorAction Stop | `
+                    where { $_.DestinationPrefix -eq $DestinationPrefix }
+
+                # Remove any default routes on the specified interface -- it is important to do
+                # this *before* removing the IP address, particularly in the case where the IP
+                # address was auto-configured by DHCP
+                if($defaultRoutes)
                 {
-                    # Remove any default routes on the specified interface -- it is important to do
-                    # this *before* removing the IP address, particularly in the case where the IP
-                    # address was auto-configured by DHCP
-                    if($defaultRoutes)
-                    {
-                        $defaultRoutes | Remove-NetRoute -confirm:$false -ErrorAction Stop
-                    }
-
-                    # Remove any IP addresses on the specified interface
-                    if($currentIP)
-                    {
-                        $currentIP | Remove-NetIPAddress -confirm:$false -ErrorAction Stop
-                    }
+                    $defaultRoutes | Remove-NetRoute -confirm:$false -ErrorAction Stop
                 }
-                else
+
+                # Remove any IP addresses on the specified interface
+                if($currentIP)
                 {
-                    # If the desired IP or default gateway is present, remove it even if NoReset
-                    # was requested; this is required if one of the settings is correct but others
-                    # are not -- otherwise, New-NetIPAddress will throw an error.
-                    if($filterGateway)
-                    {
-                        $filterGateway | Remove-NetRoute -Confirm:$false
-                    }
-
-                    if($filterIP)
-                    {
-                        $filterIP | Remove-NetIPAddress -confirm:$false -ErrorAction Stop
-                    }
+                    $currentIP | Remove-NetIPAddress -confirm:$false -ErrorAction Stop
```

Comment:
Piping is inherantly slower, could we change this to not use a pipe?

Please also captialize all parameters (ie: -Confirm:$false)